subprojects/vutils/include/vutils/vec.h

```diff
@@ -87,4 +95,9 @@
 		(to)->len += from_size;                                                            \
 	} while (0)
 
+#define VUT_VEC_FOREACH(arr, idx, var)                                                             \
+	for (size_t(idx) = 0, _capture_var = 1; (idx) < (arr)->len;                                \
+	     (idx)++, _capture_var = !_capture_var)                                                \
+		for (var = &(arr)->base[(idx)]; _capture_var; _capture_var = !_capture_var)
```

Comment:
This could be rewritten as

```suggestion
	for (size_t(idx) = 0, _capture_var = 1; (idx) < (arr)->len; (idx)++, _capture_var = 1)     \
		for (var = &(arr)->base[(idx)]; _capture_var; _capture_var = 0)
```